docs/zero_knowledge.md

```diff
@@ -1,136 +1,76 @@
+verifier cannot act maliciously. After computing and sending a ZKP, the verifier (or an eavesdropper) only learns that the prover is in possession of $w$, nothing more. This fact has the following implications:
+
+- The verifier is not able to prove to a third party that he knows $w$ (because he does not know it). 
+- The verifier is not able to prove to a third party that the initial prover knows $w$.
+
+The second point might sound a bit strange at first, so let us look at it in more detail:
+
+Say you recorded the proving procedure of the coloring problem explained above, so you can later convince your very skeptical friends that I know a valid solution. On the recording they would see that every edge you chose indeed connects two vertices of different color. But it won't convince them (they are very skeptical) since you and I could have played false and agreed on the vertices to be uncovered beforehand.
+
+So, where do we want to go with it? Using ZKPs you would, for example, be able to prove that you know a password without ever giving it to any potentially malicious third party or eavesdropper. As such, you could prove your identity to just one person and he would not be able to prove that to anyone else, nor impersonate you.  
 
+In the context of Blockchains, where data about transactions between parties is publicly stored in a shared ledger between members of the network, ZKPs can become a useful approach to itnegrate: you could prove that a transaction is valid without revealing any information about the transaction itself, thus guranteeing your privacy. But... having to play the coloring procedure again with every member of the network is not efficient at all. Here is where a new term comes to the playground: *Non-interactive ZKPs* (NIZKPs).
+
+Now, using NIZKPs, instead of having to exchange multiple messages between $P$ and $V$, the former is able to prove knowledge of $w$ by sending just a single message. This is possible thanks to different approaches, like agreeing on a *Common Reference String* (CRS) [[2]](#2), or using the Fiat-Shamir heuristic [[3]](#3).
+
+We can see our NIZKP system as a blackbox with two sets of inputs, the statement $u$ being a set of public intputs, and the witness $w$ the set of private inputs. Such a blackbox generates a proof $\pi$ that $V$ will verify, as depicted here:
+<p align=center>
+<img src="images/zkp.png" width="50%"/>
+</p>
+
+## ZKP constructions
+
+Over time, different ZKP constructions have been designed. They have beautiful names like *STARK* (Scalable Transparent ARgument of Knowledge), *Bulletproof*, *SNARG* (Succinct Non-interactive ARGuments) and *SNARK* (Succinct Non-interactive ARgument of Knowledge). I will not explain the different protocols in detail here. Instead, I will just give a brief overview of what those fancy abbreviations actually mean so you have a starting point for the next internet research rabbit hole.
```

Comment:
I know that this is something I wrote, but maybe we should use 'we' instead of 'I' here?